packages/syntax-core/src/RichSelect/RichSelectSection.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { forwardRef, type ReactElement } from "react";
+import {
+  Section as ReactAriaSection,
+  Header as ReactAriaHeader,
+  Collection as ReactAriaCollection,
+} from "react-aria-components";
+import classNames from "classnames";
+import boxStyles from "../Box/Box.module.css";
+import layoutStyles from "../layout.module.css";
+
+type RichSelectSectionProps = {
+  "data-testid"?: string;
+  label: string;
+  children: ReactElement | ReactElement[];
+  orientation?: "horizontal" | "vertical";
+};
+
+export default forwardRef<HTMLDivElement, RichSelectSectionProps>(
+  function RichSelectSection(props, ref): ReactElement {
+    const {
+      "data-testid": dataTestId,
+      label,
+      children,
+      orientation = "horizontal",
+    } = props;
+
+    return (
+      <ReactAriaSection
+        ref={ref}
+        data-testid={dataTestId}
+        className={classNames(
+          [boxStyles.box, boxStyles.flex, boxStyles.flexWrap, boxStyles.gap3],
+          {
+            [boxStyles.row]: orientation === "horizontal",
+            [boxStyles.column]: orientation === "vertical",
+          },
+        )}
+      >
+        <ReactAriaHeader className={classNames(layoutStyles.fullWidth)}>
+          {label}
```

Comment:
@benjroy NIT: can we fix the styling for these? e.g. use `Typography`

Reply:
Yep, that looks better:

<img width="672" alt="image" src="https://github.com/Cambly/syntax/assets/1890801/84c7b6a4-587d-413d-bee8-023d9647bc32">

Reply:
(change went in with this commit: https://github.com/Cambly/syntax/pull/307/commits/e9ba5968526599fbb601aebf468a34a63e18d143)